perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ExtSourcesManagerBlImpl.java

```diff
@@ -62,67 +74,111 @@ public ExtSourcesManagerBlImpl(ExtSourcesManagerImplApi extSourcesManagerImpl) {
 	@Override
 	public void initialize(PerunSession sess) {
 		if (!this.initialized.compareAndSet(false, true)) return;
+		ExtSourceGoogle.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceVOOT.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceUnity.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceCSV.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceISMU.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceLdap.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceSql.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourcePerun.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceREMS.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceTCS.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceSqlComplex.setPerunBlImpl((PerunBlImpl) perunBl);
+		ExtSourceXML.setPerunBlImpl((PerunBlImpl) perunBl);
```

Comment:
Are you sure it can't be done in more general way? To prevent anybody from forgetting of adding new ExtSource to method for initialization?

Reply:
It can, I have made all the classes to extend a new class ExtSourceImpl which has the method setPerunBl() that is called when a new instance is created.